src/src/PluginStructs/P144_data_struct.h

```diff
@@ -0,0 +1,48 @@
+// ----------------------------------------------------------------------------
+// P144 "Dust - PM1006(K) (Vindriktning)"
+// Header file for sensor abstraction
+// 2022 By flashmark
+// ----------------------------------------------------------------------------
+#ifndef PLUGINSTRUCTS_P144_DATA_STRUCT_H
+#define PLUGINSTRUCTS_P144_DATA_STRUCT_H
+
+#include "../../_Plugin_Helper.h"
+#ifdef USES_P144
+#define PLUGIN_144_DEBUG  true   // set to true for extra log info
+
+// States for statemachine used to decode received message
+typedef enum {
+    PM1006_HEADER,
+    PM1006_LENGTH,
+    PM1006_DATA,
+    PM1006_CHECK
+} pm1006_state_t;
+
+const int P144_bufferSize = 20;
+struct P144_data_struct : public PluginTaskData_base {
+  P144_data_struct() = default;
+  virtual ~P144_data_struct() = default;
+  bool     setSerial(ESPEasySerialPort portType, int8_t rxPin, int8_t txPin);  // setup serial port connected to the sensor
+  void     disconnectSerial();  // Disconnect the serial port connected to the sensor
+  bool     processSensor();     // Process received characters from sensor, must be called regularly
+  int      getValue();          // Get the last value received from the senso
+
+  private:
+  bool     processRx(char c);   // Handle one received character according to protocol
+  void     dump();              // Diagnostics, dump the serialRxBuffer
+  char*    toHex(char c, char * ptr);    // Print a int in hexadecimal
+
+  ESPeasySerial *easySerial = nullptr;    // Setial port object
+  char serialRxBuffer[P144_bufferSize];   // Receive buffer for serial RX characters
```

Comment:
You can initialize an array to 0 like this:
```c++
char serialRxBuffer[P144_bufferSize] = {0};
```
N.B. Since recent C++ standards you can also initialize other types like float using the empty `{}` notation. Both for single members as well as arrays.

Reply:
Will be fixed